component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateServlet.java

```diff
@@ -69,6 +71,21 @@ protected void doPost(HttpServletRequest servletRequest, HttpServletResponse ser
             throw new RuntimeException("Error occurred while building URL.", e);
         }
 
+        String authenticationEndpoint = X509CertificateUtil.getX509Parameters().get(X509CertificateConstants
+                .AUTHENTICATION_ENDPOINT_PARAMETER);
+
+        if (authenticationEndpoint != null) {
+            URL authenticationEndpointURL = new URL(authenticationEndpoint);
+            String protocol = authenticationEndpointURL.getProtocol();
+            String host = authenticationEndpointURL.getHost();
+            int port = authenticationEndpointURL.getPort();
+            if (port == -1) {
+                commonAuthURL = protocol + "://" + host + "/" + X509CertificateConstants.COMMON_AUTH;
+            } else {
+                commonAuthURL = protocol + "://" + host + ":" + port + "/" + X509CertificateConstants.COMMON_AUTH;
```

Comment:
Can we move `://`, `:` and `/` to constants?

Reply:
Resolved with 19a2fddeacdc3ffdeae1d0ad4429a6be8bb576de